modules/installation-mirror-repository.adoc

```diff
@@ -22,38 +22,41 @@ Complete the following steps on the bastion host:
 . Review the
 link:https://access.redhat.com/downloads/content/290/[{product-title} downloads page]
 to determine the version of {product-title} that you want to install.
+The image release advisory includes the pullspec repository and digest. For example, the link:https://access.redhat.com/errata/product/290/ver=4.3/rhel---7/x86_64/RHBA-2020:0062[put the title of the page that loads] release uses `quay.io/openshift-release-dev/ocp-release` for the pullspec repository and  `sha256:3a516480dfd68e0f87f702b4d7bdd6f6a0acfdac5cd2e9767b838ceede34d70d` for the x86-64 4.3.0 release image digest.
```

Comment:
If both image tag and image digest is showing on the same advisory page, why here we have to replace image tag with image digest. We just need tell user to get correct image tag form advisory page, no need to modify the following detailed steps. Do I miss something?

Reply:
> We just need tell user to get correct image tag form advisory page, no need to modify the following detailed steps.

Then you have to worry about divergence between the advisory and actual tag names.  Obviously we'd like to keep those synchronized, but didn't always do so over the multi-arch transition.  Also, users got used to the tag == version pattern we had been using for amd64 and were surprised when arch suffixes broke that pattern.  By-digest flows avoid implying a tag-name pattern.  Personally, I'd rather remove tag names from Errata text entirely, and this change sets the stage for that too.

Reply:
Tag name is easier reading, and I saw this in another PR - https://github.com/openshift/openshift-docs/pull/19380.

Reply:
> Tag name is easier reading...

Do you need to read this?  You just copy/paste out of the errata.

Reply:
If I am a customer, I want to get know which version my cluster is running on, which version my cluster will go forward to. But digest can not clearly tell me that. Of course, this is only my personal thoughts and preference.

Reply:
And this change make me remember my original testing mentioned in https://issues.redhat.com/browse/CORS-1105?focusedCommentId=13851937&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13851937. Do you means this change will not cause that issue any more?

Reply:
> ...But digest can not clearly tell me that...

You just copied the digest off a specific release's errata page.  So you should remember for the duration of the copy/paste.

> And this change make me remember my original testing...

Copying out here, your issue seems to have been:

```console
# oc adm upgrade --to-image=registry.svc.ci.openshift.org/ocp/release@sha256:ce9cc2ba29ab270091266f1f20240cfe6e6dcb11ccc61d5c717ca6b812925613 --force
Updating to release image registry.svc.ci.openshift.org/ocp/release@sha256:ce9cc2ba29ab270091266f1f20240cfe6e6dcb11ccc61d5c717ca6b812925613
# oc get clusterversion
NAME      VERSION                             AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.2.0-0.nightly-2019-08-26-202352   True        False         8m55s   Cluster version is registry.svc.ci.openshift.org/ocp/release@sha256:ce9cc2ba29ab270091266f1f20240cfe6e6dcb11ccc61d5c717ca6b812925613
```

First, in case anyone is tempted to copy/paste, that should use `--allow-explicit-upgrade` and not `--force`.  About the digest in the status, seems like a CVO bug to me.  As you can see from the version column, the CVO is able to retrieve the version string compiled into the release image.  And that compiled-in version is covered by the image signature, so it's more reliable than the tag, which is not.  Would addressing the status-string issue in the CVO cover your concerns with this doc change?

Reply:
>  Would addressing the status-string issue in the CVO cover your concerns with this doc change?

Yeah, if the status string can clearly show version info too, that would be better.

Reply:
In [4.6.0-0.nightly-2020-09-28-125756 -> 4.6.0-0.nightly-2020-09-28-171716][1], [the e2e logs][2] have:

```
Sep 28 18:04:05.430: INFO: Starting upgrade to version= image=registry.svc.ci.openshift.org/ocp/release:4.6.0-0.nightly-2020-09-28-171716
STEP: continuously hitting pods through the service's LoadBalancer
Sep 28 18:04:35.732: INFO: cluster upgrade is Progressing: Working towards 4.6.0-0.nightly-2020-09-28-171716: 0% complete
```

And the final ClusterVersion includes:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade/1310630479774355456/artifacts/e2e-aws-upgrade/clusterversion.json | jq '.items[] | {spec: {desiredUpdate: .spec.desiredUpdate}, status}'
{
  "spec": {
    "desiredUpdate": {
      "force": true,
      "image": "registry.svc.ci.openshift.org/ocp/release:4.6.0-0.nightly-2020-09-28-171716",
      "version": ""
    }
  },
  "status": {
    "availableUpdates": null,
    "conditions": [
      {
        "lastTransitionTime": "2020-09-28T18:00:46Z",
        "message": "Done applying 4.6.0-0.nightly-2020-09-28-171716",
        "status": "True",
        "type": "Available"
      },
      {
        "lastTransitionTime": "2020-09-28T18:00:46Z",
        "status": "False",
        "type": "Failing"
      },
      {
        "lastTransitionTime": "2020-09-28T18:13:49Z",
        "message": "Cluster version is 4.6.0-0.nightly-2020-09-28-171716",
        "status": "False",
        "type": "Progressing"
      },
      {
        "lastTransitionTime": "2020-09-28T17:31:47Z",
        "message": "Unable to retrieve available updates: currently reconciling cluster version 4.6.0-0.nightly-2020-09-28-171716 not found in the \"stable-4.6\" channel",
        "reason": "VersionNotFound",
        "status": "False",
        "type": "RetrievedUpdates"
      }
    ],
    "desired": {
      "image": "registry.svc.ci.openshift.org/ocp/release:4.6.0-0.nightly-2020-09-28-171716",
      "version": "4.6.0-0.nightly-2020-09-28-171716"
    },
    "history": [
      {
        "completionTime": "2020-09-28T18:13:49Z",
        "image": "registry.svc.ci.openshift.org/ocp/release:4.6.0-0.nightly-2020-09-28-171716",
        "startedTime": "2020-09-28T18:04:33Z",
        "state": "Completed",
        "verified": false,
        "version": "4.6.0-0.nightly-2020-09-28-171716"
      },
      {
        "completionTime": "2020-09-28T18:00:46Z",
        "image": "registry.svc.ci.openshift.org/ocp/release@sha256:5721030240b241996401f633506b91c7ad1f61acb0c45b669489df3b3917cc8c",
        "startedTime": "2020-09-28T17:31:47Z",
        "state": "Completed",
        "verified": false,
        "version": "4.6.0-0.nightly-2020-09-28-125756"
      }
    ],
    "observedGeneration": 2,
    "versionHash": "WtyYPY_PXQ0="
  }
}
```

So I think the `--to-image` version name handling was fixed by openshift/cluster-version-operator#419.

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade/1310630479774355456
[2]: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade/1310630479774355456/artifacts/e2e-aws-upgrade/container-logs/test.log